unfoldingWord/translationCore
# AlignmentData save/load to fs

#### This pull request addresses:

- Data persistence for alignmentData.
  - saves to `.apps/translationCore/alignmentData/[chapter].json`
  - loads from filesystem if exists
  - generates alignmentData if isn't in fs
    - generates at verse level for maintainability and future needs
    - empty alignments from bhp
    - wordBank from the targetLanguage

Helpers and tests to pivot alignmentData to targetLanguageVerse and back, to be used in the future for import/export to USFM 3.

#### How to test this pull request:

- Review Tests and use cases covered.
- `npm test` to ensure all tests pass
- Open any Titus project and start aligning random verses.
- Reload app and ensure that it loads the alignment data by prepopulating the UI.

- Known issues
  - There is a strange bug when clicking the top menu to switch tools and clicking on the Word Alignment tool again. It does work if you switch projects, just not tools to the same tool.


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/unfoldingword-dev/translationcore/2822)
